examples/diagram.dx

```diff
@@ -0,0 +1,193 @@
+def listLength ((AsList n _):List a) : Int = n
+
+-- TODO: we want this for any monoid but this implementation won't work.
+def concat (lists:n=>(List a)) : List a =
+  totalSize = sum for i. listLength lists.i
+  AsList _ $ fst $ withState 0 \listIdx.
+    fst $ withState 0 \eltIdx.
+      for i:(Fin totalSize).
+        while (\(). get eltIdx >= listLength (lists.((get listIdx)@_))) \().
+          eltIdx := 0
+          listIdx := get listIdx + 1
+        (AsList _ xs) = lists.((get listIdx)@_)
+        eltIdxVal = get eltIdx
+        eltIdx := eltIdxVal + 1
+        xs.(eltIdxVal@_)
+
+
+Point : Type = (Float & Float)
+
+data Geom =
+  Spot
+  Circle Float
+  Rectangle Float Float  -- width, height
```

Comment:
Why a comment and not a record type?

Reply:
That would work well too. I think I just hadn't yet become familiar with @danieldjohnson 's awesome record implementation when I wrote it.